x-pack/legacy/plugins/infra/public/components/metrics_explorer/chart.tsx

```diff
@@ -58,9 +58,12 @@ export const MetricsExplorerChart = injectUICapabilities(
       const handleTimeChange = (from: number, to: number) => {
         onTimeChange(moment(from).toISOString(), moment(to).toISOString());
       };
-      const dateFormatter = useCallback(
-        niceTimeFormatter([first(series.rows).timestamp, last(series.rows).timestamp]),
-        [series, series.rows]
+      const dateFormatter = useMemo(
```

Comment:
Should this be `useMemo` or `useCallback`?

Reply:
I proposed to use `useMemo` because we need to conditionally exchange the callback. It could also be written with `useCallback` with the difference that the condition would be evaluated on every call instead of once.

Reply:
Oh ok, I think I understand now. Memoizing the fork between `series.rows` having items or not ...